### PR TITLE
Refactor replicas

### LINKED ISSUE #1117: [BUG]Refactor replicas
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
Signed-off-by: cheyang <cheyang@163.com>

### Ⅱ. Does this pull request fix one issue?

fixes #1117 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ctrl/ctrl.go [hunk 1]
```diff
@@ -110,3 +110,59 @@ func (e *Helper) SetupWorkers(runtime base.RuntimeInterface,
...
+				utils.UpdateRuntimeCondition(statusToUpdate.Conditions,
+					cond)
+		}
+
+		if !reflect.DeepEqual(currentStatus, statusToUpdate) {
+			err = e.client.Status().Update(context.TODO(), runtime)
+		}
+
+	} else {
+		e.log.V(1).Info("No need to update runtime status for checking healthy")
```
Comment (yangyuliufeng) [inline pkg/ctrl/ctrl.go:164]: why sometimes using V(1) and sometimes not

**Review by yangyuliufeng — APPROVED** [review]
LGTM, just some little problems

**Review by yangyuliufeng — CHANGES_REQUESTED** [review]
LGTM, just some little problems

File: pkg/ctrl/ctrl.go (hunk 1 above)
Comment (cheyang) [reply pkg/ctrl/ctrl.go:164]: Using V(1) as debug info. no needs to print out in most of time.

File: pkg/utils/kubeclient/statefulset.go [hunk 2]
```diff
@@ -87,3 +89,19 @@ func getParentName(pod *v1.Pod) string {
 func isMemberOf(sts *appsv1.StatefulSet, pod *v1.Pod) bool {
 	return getParentName(pod) == sts.Name
 }
+
+// GetPhaseFromStatefulset gets the phase from statefulset
+func GetPhaseFromStatefulset(replicas int32, sts appsv1.StatefulSet) (phase datav1alpha1.RuntimePhase) {
+	if sts.Status.ReadyReplicas > 0 {
+		if replicas == sts.Status.ReadyReplicas {
+			phase = datav1alpha1.RuntimePhaseReady
+		} else if sts.Status.ReadyReplicas >= 1 {
```
[the remark this answers is not in the document]
Comment (cheyang) [reply pkg/utils/kubeclient/statefulset.go:98]: Do you mean

```golang
if sts.Status.ReadyReplicas > 0 {
		if replicas == sts.Status.ReadyReplicas {
			phase = datav1alpha1.RuntimePhaseReady
		} else {
```

File: pkg/ctrl/ctrl.go (hunk 1 above)
Comment (yangyuliufeng) [reply pkg/ctrl/ctrl.go:163]: > Using V(1) as debug info. no needs to print out in most of time.

get,thx

**Review by yangyuliufeng — APPROVED** [review]
LGTM

**Review by TrafalgarZZZ — APPROVED** [review]
LGTM

**Pull request merged**